[nccl-pg] Migrate to getCvar* functions for env variable checking

pavanbalaji opened the pull request:
Summary:
The getCvar* functions allow us to provide multiple environment variables for the same value.  This allows us to deprecate some variables in favor of others, while still allowing users to temporarily use the old variables for some time.

Test Plan: OSS CI

Reviewed By: fduwjj, XilunWu

Differential Revision: D51225487

Fixes #ISSUE_NUMBER

Skylion007 commented on torch/csrc/distributed/c10d/Utils.hpp:
```diff
@@ -87,50 +85,99 @@ inline std::vector<std::string> split(
+inline std::string getCvarString(const std::vector<std::string>& env, const char *def) {
+  const char *ret = def;
+
+  if (env.size() == 0) {
```
```suggestion
  if (env.empty()) {
```

fduwjj reviewed (approved):
Please ensure all CI pass.

XilunWu commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -39,6 +39,22 @@ namespace c10d {
+    "NCCL_USE_TENSOR_REGISTER_ALLOCATOR_HOOK"};
+
 DebugInfoWriter::DebugInfoWriter(int rank) {
   const char* fileName = parseEnvVarString(
```
need to replace this as well

pavanbalaji replied:
This patch does not rename the variables yet.  It simply introduces the new getCvar functions.  I have another commit that makes the actual changes to the env variables, which I have not pushed yet.

XilunWu reviewed (changes requested):
we can merge once CI signals are all green ;-) Hope you enjoyed the dev experience.

pavanbalaji committed:
[nccl-pg] Migrate to getCvar* functions for env variable checking (#113797)

Summary:
The getCvar* functions allow us to provide multiple environment variables for the same value.  This allows us to deprecate some variables in favor of others, while still allowing users to temporarily use the old variables for some time.


Test Plan: OSS CI

Reviewed By: fduwjj, XilunWu

Differential Revision: D51225487

Pulled By: pavanbalaji
